doc/source/force-constants.rst

```diff
@@ -178,7 +190,45 @@ Some information must be provided as a pint ``Quantity`` with both a magnitude a
 For more details on the shape of each array, see the docstrings for each object.
 An example is shown below, assuming that the inputs are being loaded from existing ``.npy`` files:
 
-.. code-block:: py
+.. testsetup:: nacl_fc_convert_npy
+
+  import numpy as np
+  import json
+  from euphonic import ForceConstants
+
+  # Create ForceConstants constructor input files
+  fc_attrs = ['cell_vectors', 'atom_r', 'atom_type', 'atom_mass',
+              'force_constants', 'sc_matrix', 'cell_origins',
+              'born', 'dielectric']
+  fnames = [attr + '.npy' for attr in fc_attrs]
+  fc = ForceConstants.from_phonopy(path=get_phonopy_path('NaCl'),
```

Comment:
This is a `testsetup` block rather than a `testcode` block (so this is just setup that is not visible to the user, and is done before `testcode` is executed). The idea here is that I'm reading from the test data in `'tests_and_analysis\test\data\phonopy_files\NaCl'` to create a `ForceConstants` object, from which I can create the other files I need in the `testcode:: nacl_fc_convert_npy` block below. The `testcode:: nacl_yaml` block above uses just 'NaCl' because I have previously copied the required files over in the `testsetup:: nacl_yaml` above it. This is to make the examples in the docs look nicer and simpler for the user, while still being able to test them.

If this block was wrong, the doc tests would (hopefully!) fail.